### PR TITLE
Feature/usb stream support

### PR DESCRIPTION
**Pull request by alibukharai**
PTAL

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,38 +1,62 @@
...
+* Support volume, mute and other features control through the UAC Control interface
+* Support stream separately suspend and resume
 
 ## Supported Drivers
 
 |                             **Driver**                             | **Version** |
 | ------------------------------------------------------------------ | ----------- |
-| [xxx](https://components.espressif.com/components/espressif/xxx) |        |
-
-## Dependencies Version
```
Comment (Lzw655) [inline README.md:19]: Are there some reasons why we should remove this section?

File: .pre-commit-config.yaml
Comment (Lzw655) [inline .pre-commit-config.yaml:1]: Are there some reasons why we should remove this file?

File: check_copyright_config.yaml
Comment (Lzw655) [inline check_copyright_config.yaml:1]: Are there some reasons why we should remove this file?

File: examples/GettingStartUAC/GettingStartUAC.ino [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-FileCopyrightText: 2023-2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Comment (Lzw655) [inline examples/GettingStartUAC/GettingStartUAC.ino]:
```suggestion
#include <Arduino.h>
```

File: examples/GettingStartUAC/GettingStartUAC.ino [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+// The loop function runs repeatedly
+void loop()
+{
+    // Delay the task for 100ms
+    vTaskDelay(5000);
+}
+
+/* Define the Mic frame callback function implementation */
+static void mic_frame_cb(mic_frame_t *frame, void *ptr)
```
Comment (Lzw655) [inline examples/GettingStartUAC/GettingStartUAC.ino]:
```suggestion
static void onMicFrameCallback(mic_frame_t *frame, void *ptr)
```

File: examples/GettingStartUVC/GettingStartUVC.ino [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    // free(_frameBuffer);
+}
+
+void loop()
+{
+    vTaskDelay(100);
+}
+
+/* Define the camera frame callback function implementation */
+static void cameraFramecb(uvc_frame *frame, void *user_ptr)
```
Comment (Lzw655) [inline examples/GettingStartUVC/GettingStartUVC.ino]:
```suggestion
static void onCameraFrameCallback(uvc_frame *frame, void *user_ptr)
```

File: examples/GettingStartUVC/GettingStartUVC.ino [hunk 5]
```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-FileCopyrightText: 2023-2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "USB_STREAM.h"
+
```
Comment (Lzw655) [inline examples/GettingStartUVC/GettingStartUVC.ino]:
```suggestion
#include <Arduino.h>
```

File: library.properties [hunk 6]
```diff
@@ -1,10 +1,9 @@
...
-category=Other
-architectures=esp32
-url=[xxx]
-includes=[xxx]
+maintainer=alibukharai
+sentence=ESP32_USB_STREAM is a specialized library created to facilitate the implementation of USB stream functionality on ESP SoCs.
+paragraph=This means that it provides a convenient and efficient way to transmit audio and video data through USB connections, making it an invaluable tool for a wide range of applications such as audio and video streaming, data transfer, and more.
+category=Communication
+url=https://github.com/esp-arduino-libs/ESP32_USB_Stream
+architectures=esp32S2, esp32S3
```
Comment (Lzw655) [inline library.properties]: Are you sure we can use `esp32S2` and `esp32S3` here? I only know `esp32` is ok.

And I think you will find its examples as `INCOMPATIBLE` in Arduino IDE.

**Review by Lzw655 — COMMENTED** [review]
Good job. Just few questions and leave some comments, PTAL.

File: README.md (hunk 1 above)
Comment (alibukharai) [reply README.md:19]: USB stream doesnot depend on anyother component. All the original files and USB related driver files are included in the path

File: .pre-commit-config.yaml
Comment (alibukharai) [reply .pre-commit-config.yaml:1]: my mistake. readded this file

File: check_copyright_config.yaml
Comment (alibukharai) [reply check_copyright_config.yaml:1]: my mistake. readded this file

File: examples/GettingStartUAC/GettingStartUAC.ino (hunk 3 above)
Comment (alibukharai) [reply examples/GettingStartUAC/GettingStartUAC.ino]: done

File: examples/GettingStartUVC/GettingStartUVC.ino (hunk 4 above)
Comment (alibukharai) [reply examples/GettingStartUVC/GettingStartUVC.ino]: done

File: examples/GettingStartUVC/GettingStartUVC.ino (hunk 5 above)
Comment (alibukharai) [reply examples/GettingStartUVC/GettingStartUVC.ino]: done

File: library.properties (hunk 6 above)
Comment (alibukharai) [reply library.properties]: esp32 dont have usb. only esp32s2 and esp32s3 is ok

File: README.md [hunk 7]
```diff
@@ -1,38 +1,70 @@
...
+
+usb->start();
+
+/*Dont forget to free the allocated memory*/
+// free(_xferBufferA);
+// free(_xferBufferB);
+// free(_frameBuffer);
+
 ```
+
```
Comment (Lzw655) [inline README.md]: Remove unnecessary blank lines?

File: library.properties [hunk 8]
```diff
@@ -1,10 +1,8 @@
-name=[xxx]
+name=ESP32_USB_STREAM
 version=0.0.1
 author=espressif
-maintainer=[xxx]
-sentence=[xxx] is a library designed for [xxx] using ESP SoCs
-paragraph=Currently support [xxx]
-category=Other
-architectures=esp32
```
Comment (Lzw655) [inline library.properties:8]: Miss this one?

File: library.properties [hunk 9]
```diff
@@ -1,10 +1,8 @@
...
-url=[xxx]
-includes=[xxx]
```
Comment (Lzw655) [inline library.properties:10]: And this.

File: src/USB_STREAM.cpp [hunk 10]
```diff
@@ -0,0 +1,338 @@
+/*
+ * SPDX-FileCopyrightText: 2023-2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "USB_STREAM.h"
+#include "original/include/usb_stream.h"
+#include "freertos/event_groups.h"
+#include <Arduino.h>
```
Comment (Lzw655) [inline src/USB_STREAM.cpp]: Need it?

**Comment by joyel24** [comment]
23 files changed, +8,131 lines of code, PR comment :
"Please Take A Look" 
Excellent !👌😁

**Pull request merged**